### PR TITLE
Feat/page list rides

### PR DESCRIPTION
**Pull request by Scttpr**

### CODE REVIEW

**Comment by Scttpr** [comment]
[Historique de courses](https://trello.com/c/pf71hiNl/209-historique-de-courses)

File: pages/my-account/current-rides.vue [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  data() {
+    return {
+      isModalActive: false,
+    }
+  },
+  async asyncData({ $api, $auth, $toast}) {
+    let data;
+    try {
+      data = await filterManager.fetchDatas(
+        $api.rides(null, mask).getRides,
```
Comment (Nainterceptor) [inline pages/my-account/current-rides.vue]: null as first param mean without start or end date ? If yes, it's not wanted :/

File: pages/my-account/my-rides.vue [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        </ul>
+      </div>
+    </div>
+  </section>
+</template>
+
+<script>
+import RideCard from '~/components/ride-card';
+import FilterDropdown from '~/components/elements/filter-dropdown';
+import FilterManager from '../../helpers/FilterManager';
```
Comment (Nainterceptor) [inline pages/my-account/my-rides.vue]: You can use relative `~`

File: pages/my-account/my-rides.vue [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+const filterManager = new FilterManager('fr-FR', 2019);
```
Comment (Nainterceptor) [inline pages/my-account/my-rides.vue]: Please rely to Luxon locale

File: pages/my-account/my-rides.vue [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  components: {
+    RideCard,
+    FilterDropdown,
+  },
+  async asyncData({ $api, $auth, $toast }) {
+    let data;
+    const { start, end } = filterManager.getFilter(currents);
+    try {
+      data = await filterManager.fetchDatas(
+        $api.rides(null, mask).getRides,
```
Comment (Nainterceptor) [inline pages/my-account/my-rides.vue]: Same thing about null as first param

File: pages/my-account/my-rides.vue [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        this.rides = formatData(data);
+      } catch (err) {
+        this.$toast.error(err);
+      }
+    }
+  }
+};
+</script>
+
+<style lang="scss" scoped>
```
Comment (Nainterceptor) [inline pages/my-account/my-rides.vue]: Useless block

File: helpers/FilterManager.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      years.push(year);
+    }
+    for (let month = 0; month < 12; month += 1 ) {
+      months.push(new Date(null, month).toLocaleString(locale, { month: 'long'}));
+    }
+    this.locale = locale;
+    this.years = years;
+    this.months = months;
+    this.currentYear = new Date().getFullYear().toString();
+    this.currentMonth = new Date().toLocaleString(locale, { month: 'long' });
```
Comment (Nainterceptor) [inline helpers/FilterManager.js]: Aha, if you're importing Luxon, you can use Luxon to format data
Here, I think you can do

```js
import { DateTime } from 'luxon';
//...
DateTime.local().year; // for this.currentYear
DateTime.local().toLocaleString({ month: 'long' }); // for this.currentMonth
```
Luxon will rely directly on default Locale.
You have also a lot of tools to start from current date then get the first day of the month, or things like this.

File: helpers/FilterManager.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  getCurrents() {
+    return {
+      year: this.currentYear,
+      month: this.currentMonth,
+    }
+  }
+
+  getFilter({ year, month }) {
+    const monthIndex = this.months.findIndex(currentMonth => currentMonth === month) + 1;
+    const maxDays = new Date(year, monthIndex, 0).getDate();
```
Comment (Nainterceptor) [inline helpers/FilterManager.js]: Same thing, we've Luxon in the project, use it ;)

File: helpers/FilterManager.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    return {
+      start: `${year}-${monthIndex}-1`,
+      end: `${year}-${monthIndex}-${maxDays}`,
+    };
+  }
+
+  formatDate(date) {
+    const day = new Date(date).toLocaleString(this.locale, { weekday: 'long', day: '2-digit' });
+    const hour = new Date(date).toLocaleString(this.locale, { hour: '2-digit', minute: '2-digit' });
```
Comment (Nainterceptor) [inline helpers/FilterManager.js]: Here too, We've Luxon, use it !

File: pages/my-account/current-rides.vue [hunk 9]
```diff
@@ -0,0 +1,106 @@
+<template>
+  <div>
+    <div class="columns">
+      <div class="column is-5">
+        <p class="alert-message" v-if="rides !== null || rides.length < 1">
```
Comment (Nainterceptor) [inline pages/my-account/current-rides.vue]: I'm pretty sure that this code will fail if `rides = null`, is it `||` or `&&` ?

File: pages/my-account/current-rides.vue [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+      @toggle-modal="toggleModal"
+      @action="deleteRide"
+    />
+  </div>
+</template>
+
+<script>
+import Modal from '~/components/modal'
+import RideCard from '~/components/ride-card';
+import FilterManager from '../../helpers/FilterManager';
```
Comment (Nainterceptor) [inline pages/my-account/current-rides.vue]: Please use ~

File: pages/my-account/my-rides.vue [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+            :current="currents.month"
+            :list="selects.months"
+            @filter-results="setFilter"
+          />
+        </nav>
+      </div>
+      </div>
+    <div class="columns">
+      <div class="column is-5">
+        <p class="alert-message" v-if="rides === null || rides.length < 1">
```
Comment (Nainterceptor) [inline pages/my-account/my-rides.vue]: Here it's working ;p

File: pages/my-account/my-rides.vue [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+      ...ride,
+      departure: ride.departure.label,
+      arrival: ride.arrival.label,
+      day,
+      hour,
+    };
+});
+
+export default {
+  name: 'Rides',
```
Comment (Nainterceptor) [inline pages/my-account/my-rides.vue]: Useless key

**Pull request merged**